### DISCUSSION TITLE
trainer.check

### DISCUSSION THREAD

**Discussion by gianscarpe** [Lightning-AI/pytorch-lightning#5407] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi pytorch-lightning friends! :) I'd like to start a discussion about `trainer.check` api. 
Basically, this API should check that all the user-define classes (models, data, callbacks, ...) are programmatically sound. I propose to use `inspect` to check for function correctness, here's my PR proposal at https://github.com/PyTorchLightning/pytorch-lightning/issues/3244

**Comment by carmocca** [comment, marked as answer]
Locking in favor of https://github.com/PyTorchLightning/pytorch-lightning/discussions/6029
